### PR TITLE
kernel: Multiple changes.

### PR DESCRIPTION
**Pull request by flexibeast**
* Add information about newer-than-default kernels.
* Describe how to build DKMS modules for newer-than-default kernels.
* Fix typos.

Partially addresses #337.

### CODE REVIEW

**Comment by ericonr** [comment]
Needs some more changes, if you could put them in this PR too c:

```
      ericonr │ duncaen: mind explaining what's wrong in https://github.com/void-linux/void-docs/commit/e478095076155005cc24c9cb7aecf6449979f9c4#r40050286 ? I don't think we had any complaints about that
      ericonr │ I can open an issue about it, after I know what's wrong
      duncaen │ I wanted to link to it for the ""How do I update grub" PR I just closed
      duncaen │ but its wrong
      duncaen │ xbps-query -p install-script grub
      duncaen │ first it would need to force reconfigure
      duncaen │ but also, there is nothing actually regenerating the config in grubs configure scripts
      ericonr │ I see
      ericonr │ it seems the grub kernel hook would regenerate the config, right?
      ericonr │ though I guess that's not really the correct way to go about it
      duncaen │ yes you could reconfigure a kernel package, but yea that kinda useless
      duncaen │ update-grub or what cat /usr/bin/update-grub does
```

**Comment by flexibeast** [comment]
To clarify, instead of:

> then reconfiguring the `grub` package.

it should say:

> then running:
> ```
> # grub-mkconfig -o /boot/grub/grub.cfg
> ```

? Am i understanding correctly?

**Comment by ericonr** [comment]
I think running `update-grub` is better, since it does just that. But yes, reconfiguring the grub package isn't enough.

**Comment by flexibeast** [comment]
@ericonr: Thanks, PR updated.

File: src/config/kernel.md [hunk 1]
```diff
@@ -3,16 +3,21 @@
 ## Kernel series
 
 Void Linux provides many kernel series in the default repository. These are
-named `linuxX.X`, for example `linux4.19`. You can query for all available
+named `linuxX.X`: for example, `linux4.19`. You can query for all available
```
Comment (ericonr) [inline src/config/kernel.md:6]: I'm always unsure. Should we prefer `linuxX.Y` or `linuxX.X`?

File: src/config/kernel.md [hunk 2]
```diff
@@ -126,7 +131,7 @@ repositories.
 
 Kernel command line arguments can be added through the GRUB bootloader by
 editing `/etc/default/grub`, changing the `GRUB_CMDLINE_LINUX_DEFAULT` variable
-and then reconfiguring the `grub` package.
+and then running `update-grub`.
```
Comment (ericonr) [inline src/config/kernel.md:134]: I hate that we can't link to a man page here :p

File: src/config/kernel.md (hunk 1 above)
Comment (flexibeast) [reply src/config/kernel.md:6]: Fair point - i can't see any reason *not* to change it to `X.Y`, so i'll do that.

File: src/config/kernel.md [hunk 3]
```diff
@@ -3,16 +3,21 @@
...
-kernel packages, usually the latest mainline kernel that works with all dkms
-modules.
+The `linux` meta package, installed by default, depends on one of the kernel
+packages, usually the latest mainline kernel that works with all DKMS modules.
+Newer kernels might be available in the repository, but are not necessarily
+considered stable enough to be the default; use these at your own risk. If you
+wish to use a more recent kernel and have DKMS modules that you need to build,
+install the relevant `linuxX.Y-headers` package, then use
+[xbps-reconfigure(1)](https://man.voidlinux.org/xbps-reconfigure.1) to
+reconfigure the `linuxX.Y` package you installed.
```
Comment (ericonr) [inline src/config/kernel.md:20]: Make it clearer that the reconfiguration step will build the DKMS modules? Not sure if it's needed.

Comment (flexibeast) [reply src/config/kernel.md]: i don't think it'll hurt, so i've just added "This will build the DKMS modules" to the end of that para.

**Comment by flexibeast** [comment]
@ericonr: Thanks for the review. Merging.

**Pull request merged**